jpa/org.eclipse.persistence.jpa/src/main/java/org/eclipse/persistence/jpa/metadata/FileBasedProjectCache.java

```diff
@@ -51,6 +51,7 @@ public Project retrieveProject(Map<String, ?> properties, ClassLoader loader, Se
                 java.io.File file = new java.io.File(fileName);
                 java.io.FileInputStream fis = new java.io.FileInputStream(file);
                 in = new java.io.ObjectInputStream(fis);
+                in.setObjectInputFilter(new FileBasedProjectCacheFilter(log));
```

Comment:
why is the logger needed here? if something goes wrong, exception is thrown anyway and logged few lines bellow

Reply:
Logger was there because of more complex code in first commit. It does not make sense now so I removed logger entirely.